File: spec/HelloWorld.lua

-- not sure what to do with field construction syntax...
position = LisztFieldWithLabel[Vertex,Vec[_3,Float]]("position")
field    = LisztField[Cell,Vec[_2,Int]](Vec(0,0))
field2   = LisztField[Cell,Int](0)
myid     = LisztField[Vertex,Int](0) --,Vec[_2,Int]](Vec(0,0))

function main () {
	print("hello")

	local a = LisztVec(1,2)

	mesh.cells().map(
		liszt_kernel (cell)
			print(cell)
			field(cell) = LisztVec(ID(cell), ID(cell)+ 1)

			for v in vertices(cell) do
				print(position(v))
			end
		end
	)

	mesh.cells().map(
		liszt_kernel (cell)
			print(cell, field(cell))
			print(a)
		end
	)
	print(a)

	mesh.cells().map(
		liszt_kernel (cell)
			a += LisztVec(1, 2)
		end
	)

	mesh.vertices().map(
		liszt_kernel (v)
			myid(v) += ID(v)
		end
	)

	--	...

	mesh.vertices().map(
		liszt_kernel (v)
			for c in cells(v)
				field2(c) += 1
			end
		end
	)

	-- ...

}

File: spec/HelloWorld.scala
import Liszt.Language._

@lisztcode
object Main {
  val position = FieldWithLabel[Vertex,Vec[_3,Float]]("position")
  val field = FieldWithConst[Cell,Vec[_2,Int]](Vec(0,0))
  val field2 = FieldWithConst[Cell,Int](0)
  val myid = FieldWithConst[Vertex,Int](0)//,Vec[_2,Int]](Vec(0,0))
  def main() {
  	Print("hello")
  	
  	var a = Vec(1,2)
  	for(c <- cells(mesh)) {
  		Print(c)
  		field(c) = Vec(ID(c),ID(c)+1)
  		for(v <- vertices(c))
  			Print(position(v))
  	}
  	for(c <- cells(mesh)) {
  		Print(c,field(c))
  		Print(a)
  	}
  	Print(a)
  	for(c <- cells(mesh)) {
  		a += Vec(1,2)
  	}
  	Print(a)
  	for(v <- vertices(mesh)) {
  		myid(v) += ID(v) //Vec(ID(v),ID(v)+1)
  	}
  	for(c <- cells(mesh)) {
  		for(v <- vertices(c)) {
  			Print(v," ",myid(v))
  		}
  	}
  	
  	for(v <- vertices(mesh)) {
  		for(c <- cells(v)) {
  			field2(c) += 1
  		}
  	}
  	for(c <- cells(mesh)) {
  		Print(c,field2(c))
  	}
  }
}